xarray/core/rolling_exp.py

```diff
@@ -0,0 +1,58 @@
+
+import numpy as np
+from pandas.core.window import _get_center_of_mass
+
+
+def _get_alpha(com=None, span=None, halflife=None, alpha=None):
+    # pandas defines in terms of com (converting to alpha in the algo)
+    # so use its function to get a com and then convert to alpha
+
+    com = _get_center_of_mass(com, span, halflife, alpha)
+    return 1 / (1 + com)
+
+
+def rolling_exp_nanmean(array, *, axis, window):
+    import numbagg
+    if axis == ():
+        return array.astype(np.float)
```

Comment:
I changed this from @shoyer 's suggestion of `array * 1.0` given this should be a no-op for floats and is self-explaining, but let me know if that is better

Reply:
I think `np.float64` would be preferrable here -- `np.float` is actually is an alias to the builtin `float` type :(.

I was originally thinking that `array * 1.0` did more minimal upcasting (i.e., sometimes to float32), but it does look like it's doing the same thing here.